activerecord/test/cases/tasks/postgresql_rake_test.rb

```diff
@@ -228,14 +237,16 @@ def setup
 
     def test_structure_load
       filename = "awesome-file.sql"
-      Kernel.expects(:system).with("psql -q -f #{filename} my-app-db")
+      args = [['psql', 'psql',], '-q', '-f', filename, 'my-app-db']
+      Kernel.expects(:system).with(*args).returns(true)
 
       ActiveRecord::Tasks::DatabaseTasks.structure_load(@configuration, filename)
     end
 
     def test_structure_load_accepts_path_with_spaces
       filename = "awesome file.sql"
-      Kernel.expects(:system).with("psql -q -f awesome\\ file.sql my-app-db")
+      args = [['psql', 'psql',], '-q', '-f', 'awesome\\ file.sql', 'my-app-db']
+      Kernel.expects(:system).with(*args).returns(true)
```

Comment:
Why mocking is dangerous: a play in two lines.

(Yes, I know we don't have much choice here.)